File: solutions/ruby/darts/2/darts.rb
class Darts

  TP = TARGET_POINTS = {
        ..1  => 10,
     1.01..5  => 5,
     5.01..10 => 1,
    10.01..   => 0
  }

  private_constant :TP

  private

  attr_reader :gipotenuza

  def initialize(x, y)
    @gipotenuza = Math.sqrt(x**2 + y**2).ceil
  end

  public

  def score
    TP.select { |range, point| range.cover?(gipotenuza) }.values.first || 0
  end

end

File: solutions/python/darts/2/darts.py
import math

def score(x, y):
    hypotenuse = math.sqrt(x**2 + y**2)

    if 0 <= hypotenuse <= 1:
        return 10
    if 1 < hypotenuse <= 5:
        return 5
    if 5 < hypotenuse <= 10:
        return 1
    return 0
